stellar/django-polaris
# Claimable balance: Attempt 2

# Description

Submits deposit as a claimable balance if the wallet supports claimable balances via sending "claimable_balance_supported=True" in their POST request body.
and the wallet user does not trust the asset their depositing.

If claimable_balance_supported=False or they do not sent that value in `POST /transactions/deposit/interactive` request body and the wallet user does not trust the asset their depositing
then the deposit flow is blocked via pending_trust. 
At which point the wallet user must first create a trust line to that asset in order to complete the deposit.

Addresses issue: #316 

## Type of change


# How Has This Been Tested?

- [X] pytest -c polaris/pytest.ini  
    - 1 new test `test_deposit_stellar_no_trustline_with_claimable_bal`
- [X] Submit a deposit from a wallet to Polaris (Where the user doesn't trust the asset.)

The following are steps in order to test this behavior on your end.
1. Checkout the PR branch `git checkout 316-claimable-balance-attempt2` 
1. Checkout the stellar-demo-wallet branch from my fork https://github.com/Zagan202/stellar-demo-wallet/tree/deposit-untrusted-TEST-asset `git checkout deposit-untrusted-TEST-asset`
    - This is a janky test branch that allows the user to deposit offchain value for an asset named SRT
    - Balance should look like this 
![Screen Shot 2020-11-30 at 11 26 54 AM](https://user-images.githubusercontent.com/5233187/100635824-fd702800-32fe-11eb-8a68-15bacae8a1e7.png)
1. Start the wallet `npm start`
1. Build and run Polaris (server and poll_pending_deposits)
   - docker-compose build
   - Execute in one terminal plane `docker-compose up server`
   - Execute in another terminal plane `docker-compose up poll_pending_deposits`
1. Verify that your SRT asset is setup properly 
   - Checkout your `http://localhost:8000/.well-known/stellar.toml`
   - Checkout `http://localhost:8000/adminpolaris/asset/`
1. Login to this [account](http://localhost:3333/?secretKey=SD3XK7YPIJMVUCAFIP34URDZNG3A4QUWCGW6TB4I2JHRQ7DVZCU3G4R2) 
1. Execute a SRT deposit
   - It give you the interactive window then you'll see a claimable balance available for whatever you deposited
